memory/src/main/scala/filodb.memory/data/OffheapLFSortedIDMap.scala

```diff
@@ -360,31 +537,30 @@ extends OffheapLFSortedIDMapReader(memFactory, holderKlass) {
   //scalastyle:on
 
   /**
-   * Atomically calls the elementFunc and inserts the element it returns IF AND ONLY IF
-   * the item with the given key is not already in the map.  elementFunc is not called if key already exists.
+   * Atomically inserts the element it returns IF AND ONLY IF
```

Comment:
Sorry, what was the rationale again for not being able to use a function?

Reply:
If the function is called when the lock is held, and the function decides to allocate memory, that reclaim listener could go back to the chunk map and cause a deadlock.